Dockerfile

```diff
@@ -0,0 +1,49 @@
+FROM golang:1.19.1-alpine3.16 AS builder
+
+WORKDIR /app
+RUN set -eux; \
+    apk add --no-cache \
+        gc-dev \
+        make \
+        gcc \
+        git \
+        musl-dev \
+        pcre-dev \
+    ;
+
+COPY . .
```

Comment:
It is a good idea to first copy the go.mod and go.sum, run go get ./... and then copy the rest of the code. This way you will cache the pull of deps.